File: app-frontend/lib/insights.ts
export function generateInsights(
  income: number,
  fixed: number,
  variable: number
) {
  const insights: string[] = [];
  const savings = income - fixed - variable;

  if (savings < income * 0.2) {
    insights.push(
      "Your savings rate is below 20%. Consider reducing variable expenses."
    );
  }

  if (variable > fixed) {
    insights.push(
      "You are spending more on wants than needs. Watch discretionary expenses."
    );
  }

  if (savings <= 0) {
    insights.push(
      "You are spending more than your income. Budget adjustment needed."
    );
  }

  return insights;
}

File: model-backend/services/insights.py
def generate_insights(income, fixed, variable):
    insights = []
    savings = income - fixed - variable

    if savings < income * 0.2:
        insights.append("Savings rate below 20%. Consider cutting variable expenses.")

    if variable > fixed:
        insights.append("Wants spending exceeds needs. Watch discretionary expenses.")

    if savings <= 0:
        insights.append("You are overspending. Budget correction needed.")

    return insights
